xarray/tests/test_variable.py

```diff
@@ -1681,3 +1683,75 @@ def test_raise_no_warning_for_nan_in_binary_ops():
     with pytest.warns(None) as record:
         Variable('x', [1, 2, np.NaN]) > 0
     assert len(record) == 0
+
+
+class TestBackendIndexing(TestCase):
+    """    Make sure all the array wrappers can be indexed. """
+    def setUp(self):
+        self.d = np.random.random((10, 3)).astype(np.float64)
+
+    def check_orthogonal_indexing(self, v):
+        assert np.allclose(v.isel(x=[8, 3], y=[2, 1]),
+                           self.d[[8, 3]][:, [2, 1]])
+
+    def check_vectorized_indexing(self, v):
+        ind_x = Variable('z', [0, 2])
+        ind_y = Variable('z', [2, 1])
+        assert np.allclose(v.isel(x=ind_x, y=ind_y), self.d[ind_x, ind_y])
+
+    def test_NumpyIndexingAdapter(self):
```

Comment:
FYI for the future, I think these combinations could be handled through pytest parameterization well (but within the existing framework, this looks good)

Reply:
Thanks.
Do you mean something similar to [this](https://github.com/pydata/xarray/blob/master/xarray/tests/test_dataset.py#L4063)?
If more tests are needed, I will adopt this feature.

Reply:
Yes exactly! Thanks